src/main/java/edu/ie3/datamodel/io/source/csv/CsvCoordinateSource.java

```diff
@@ -0,0 +1,130 @@
+    super(csvSep, folderPath, fileNamingStrategy);
+    this.fileNamingStrategy = fileNamingStrategy;
+    setupCoordinateMaps();
+  }
+
+  public void setupCoordinateMaps() {
+    Optional<String> fileName = fileNamingStrategy.getCoordinateFileName();
+    if (!fileName.isPresent()) {
+      log.error("Cannot read coordinates as I could not get a valid file name");
+      return;
+    }
+    Stream<Map<String, String>> fieldsToAttributes =
+        buildStreamWithFieldsToAttributesMap(fileName.get(), connector);
+    idToCoordinate = buildIdToCoordinateMap(fieldsToAttributes);
+    coordinateToId = new HashMap<>();
+    idToCoordinate.forEach((k, v) -> coordinateToId.put(v, k));
+  }
+
+  public Point getCoordinate(Integer id) {
+    return idToCoordinate.get(id);
+  }
+
+  @Override
+  public Collection<Point> getCoordinates(Integer... ids) {
+    return null;
+  }
+
+  public Collection<Point> getCoordinates(int... ids) {
+    return Arrays.stream(ids).mapToObj(this::getCoordinate).collect(Collectors.toSet());
+  }
+
+  public Collection<Point> getCoordinatesBetween(Integer fromId, Integer toId) {
+    return getCoordinates(IntStream.rangeClosed(fromId, toId).toArray());
+  }
+
+  public Integer getId(Point coordinate) {
+    return coordinateToId.get(coordinate);
+  }
+
+  public Integer getCoordinateCount() {
+    return idToCoordinate.keySet().size();
+  }
+
+  private HashMap<Integer, Point> buildIdToCoordinateMap(
+      Stream<Map<String, String>> fieldsToAttributes) {
+    HashMap<Integer, Point> coordinateMap = new HashMap<>();
+
+    fieldsToAttributes.forEach(
+        map -> {
+          try {
+            double lat = Double.parseDouble(map.get(latitudeHeader));
+            double lon = Double.parseDouble(map.get(longitudeHeader));
+            Integer id = Integer.valueOf(map.get(idHeader));
+            Point coordinate = GeoUtils.xyToPoint(lat, lon);
+            coordinateMap.put(id, coordinate);
+          } catch (NumberFormatException e) {
+            log.warn("Could not build coordinate from given values: {}", map);
+          }
+        });
+    return coordinateMap;
+  }
+
+  /**
+   * Tries to open a file reader from the connector based on the provided file name, reads the first
+   * line (considered to be the headline with headline fields) and returns a stream of (fieldName ->
+   * fieldValue) mapping where each map represents one row of the .csv file. Since the returning
+   * stream is a parallel stream, the order of the elements cannot be guaranteed.
+   *
+   * @param filename the file name that is used to get the corresponding reader
+   * @param connector the connector that should be used to get the reader from
+   * @return a parallel stream of maps, where each map represents one row of the csv file with the
+   *     mapping (fieldName -> fieldValue)
+   */
+  private Stream<Map<String, String>> buildStreamWithFieldsToAttributesMap(
```

Comment:
This is almost a 1:1 duplicate from `CsvDataSource.buildStreamWithFieldsToAttributesMap(...)` what's the reason for this duplication? I think this can be simplified. We can discuss how if you want to.

Reply:
The issue I had was with the UniqueEntity restriction. Coordinates are not mapped by us, so they are not UniqueEntities.